lib/compress/zstd_compress_advanced.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016-present, Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef ZSTD_COMPRESS_ADVANCED_H
+#define ZSTD_COMPRESS_ADVANCED_H
+
+/*-*************************************
+*  Dependencies
+***************************************/
+
+#include "zstd_internal.h"
+
+
+// #if defined (__cplusplus)
+// extern "C" {
+// #endif
+
+/*-*************************************
+*  Target Compressed Block Size
+***************************************/
+
+typedef struct {
+    symbolEncodingType_e hType;
+    BYTE hufDesBuffer[500];
+    size_t hufDesSize;
+} ZSTD_hufCTablesMetadata_t;
```

Comment:
It only matters to expose these definitions if they are going to be used by the "user" (the source file which `#include` this header).

Typically, it would be necessary if this type was required to invoke one of the following functions.

But that's not the case. Apparently these types are only useful within `zstd_compress_advanced.c`, so, following the principle of leanest scope, that's where they should be declared.